contracts/ColonyTask.sol

```diff
@@ -321,6 +321,22 @@ contract ColonyTask is ColonyStorage, DSMath {
   taskNotFinalized(_id)
   {
     tasks[_id].cancelled = true;
+
+    // Get task and domain pots so we can return the funds
+    uint256 taskPotId = tasks[_id].potId;
+    Pot storage taskPot = pots[taskPotId];
+    uint256 domainId = tasks[_id].domains[0];
+    uint256 domainPotId = domains[domainId].potId;
+    uint256 taskTokenBalance = taskPot.balance[token];
+    uint256 taskEtherBalance = taskPot.balance[0x0];
+
+    // Return task funds to the domain if a pot is not empty
```

Comment:
The task can actually be funded by a number of different (whitelisted) tokens so this approach won't scale well. Instead, after a task is cancelled, you should call `moveFundsBetweenPots` explicitly for as many types of funding there're are e.g this is a change that probably doesn't need any contract updates just a test to prove it works.